fs/ignorefs/ignorefs.go

```diff
@@ -132,6 +147,10 @@ func (d *ignoreDirectory) Readdir(ctx context.Context) (fs.Entries, error) {
 			continue
 		}
 
+		if !thisContext.shouldIncludeByDevice(e) {
```

Comment:
This looks very good, however there's something weird about having to have Parent* together with current device information., leads to some awkward construction of those nodes.

Could we simply compare `d.Device()` with `e.Device()` here? I believe this would get rid of the need to expose Parent*, because `d` is the parent.